products/llbuild-analyze/Sources/CriticalPathTool.swift

```diff
@@ -272,3 +291,119 @@ struct EncodableRuleResult: Encodable {
         try container.encode(deps, forKey: .dependencies)
     }
 }
+
+
+protocol GraphVizNode {
+    var graphVizName: String { get }
```

Comment:
Is the output format you chose a convention for GraphViz? If not, could we use the existing logic in `BuildKey.description`? That uses a xml-like representation of the object, e.g. for `BuildKey.Command`: `<BuildKey.\(type(of: self)) name=\(name)>` -> `<BuildKey.Command name=A>`.

Reply:
It's not a convention, I thought the XML-ish syntax looked a bit odd in the generated PDF.

For now, I've made the graphVizName return the description, maybe we can add options what info to include in a future version.

Reply:
The XML-ish description is not a convention either. We could also change that 🙂